pytensor/tensor/slinalg.py

```diff
@@ -67,29 +67,52 @@ def make_node(self, x):
     def perform(self, node, inputs, outputs):
         [x] = inputs
         [out] = outputs
-        try:
-            # Scipy cholesky only makes use of overwrite_a when it is F_CONTIGUOUS
-            # If we have a `C_CONTIGUOUS` array we transpose to benefit from it
-            if self.overwrite_a and x.flags["C_CONTIGUOUS"]:
-                out[0] = scipy_linalg.cholesky(
-                    x.T,
-                    lower=not self.lower,
-                    check_finite=self.check_finite,
-                    overwrite_a=True,
-                ).T
-            else:
-                out[0] = scipy_linalg.cholesky(
-                    x,
-                    lower=self.lower,
-                    check_finite=self.check_finite,
-                    overwrite_a=self.overwrite_a,
-                )
 
-        except scipy_linalg.LinAlgError:
-            if self.on_error == "raise":
-                raise
-            else:
+        # Quick return for square empty array
+        if x.size == 0:
+            eye = np.eye(1, dtype=x.dtype)
+            (potrf,) = scipy_linalg.get_lapack_funcs(("potrf",), (eye,))
+            c, _ = potrf(eye, lower=False, overwrite_a=False, clean=True)
+            out[0] = np.empty_like(x, dtype=c.dtype)
+            return
+
+        x1 = np.asarray_chkfinite(x) if self.check_finite else x
```

Comment:
I don't like the introduction of a new variable. We know for sure `x` is a numpy array inside this function, so the only relevant code in `np.asarray_chkfinite(x)` is the `if not np.isfinite(x).all(): ` part. We should just directly do that

Reply:
Do you think that the `on_error` flag should affect whether or not a `ValueError` is raised? Or should it just be:
```suggestion
        if self.check_finite and not np.isfinite(x).all():
            raise ValueError("array must not contain infs or NaNs")
```

Reply:
Yes, `on_error` should effect whether an error raises. If `on_error='nan'` there should not be an error raised, it should return a matrix of the correct shape with all `nan`